### PR TITLE
Add Devise

### PR DESCRIPTION
**Pull request by AbelG101**
## Changes
- Add user login page.
- Use devise gem for authentication.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
-
-  def recent_posts
-    posts.limit(3).order(created_at: :asc)
-  end
-end
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (sja-thedude) [inline app/models/user.rb]: - [ ] Kindly make sure to add `:confirmable` it is required that it should be enabled in the user model (user.rb file). You should have something similar to what is below:
```
devise :database_authenticatable, :registerable,
:recoverable, :rememberable, :validatable, :confirmable
```

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @AbelG101 & @DuaneDave, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ All linters passing
✔️ Professional Readme
✔️ Good job so far

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (AbelG101) [reply app/models/user.rb]: @sja-thedude Thank you for noticing, I'll make the changes.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    @current_user = User.find(params[:user_id])
+  before_action :configure_permitted_parameters, if: :devise_controller?
```
Comment (Munsa1) [inline app/controllers/application_controller.rb:2]: - [ ] While you have made a great effort in implementing the required changes, this has introduced a new bug in the codebase. I am currently unable to sign up. Upon submitting for a sign up I get the following error
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @AbelG101  @DuaneDave , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (AbelG101) [reply app/controllers/application_controller.rb:2]: @Munsa1 Please excuse my recklessness 🙊, I've double-checked everything It should work fine now.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @AbelG101, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AbelG101** [comment]
@Mihndim2020 Thank you and happy coding! 🥳🥳

**Pull request merged**